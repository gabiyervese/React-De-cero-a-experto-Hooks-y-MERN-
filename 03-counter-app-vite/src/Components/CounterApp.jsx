import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value, name, luckyNumber }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleSubs = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>Counter App</h1>
      <h2 data-testid="test-title">{name}</h2>
      <h3>{counter}</h3>
      <p>{luckyNumber}</p>
      {/* <p>{value}</p> */}
      <button onClick={handleAdd}>+</button>
      <button onClick={handleSubs}>-</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
  name: PropTypes.string.isRequired,
};

CounterApp.defaultProps = {
  // name: "Gabriela",
  luckyNumber: "6",
};
