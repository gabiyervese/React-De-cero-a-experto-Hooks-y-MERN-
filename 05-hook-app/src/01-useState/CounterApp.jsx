import { useState } from "react";

const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const { counter1, counter2, counter3 } = counter;

  return (
    <>
      <h1>Lets count</h1>
      <h2>Counter1: {counter1}</h2>
      <h2>Counter2: {counter2}</h2>
      <h2>Counter3: {counter3}</h2>
      <button
        className="btn"
        onClick={() => setCounter({ ...counter, counter1: counter1 + 1 })}
      >
        + 1
      </button>
    </>
  );
};

export default CounterApp;
