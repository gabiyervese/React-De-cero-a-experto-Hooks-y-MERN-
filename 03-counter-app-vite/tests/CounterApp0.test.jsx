const { render, screen, fireEvent } = require("@testing-library/react"); //screen es el objeto que estamos renderizando.
import { CounterApp } from "../src/Components/CounterApp";

describe("Pruebas basicas del componenete <CounterApp/>", () => {
  const luckyNumber = 6;
  const value = 18;

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<CounterApp luckyNumber={luckyNumber} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el valor inicial de 100 en el counter", () => {
    render(<CounterApp luckyNumber={luckyNumber} />);
    expect(screen.getByText(6)).toBeTruthy();
  });

  test("debe de incrementar con el boton +", () => {
    render(<CounterApp value={value} />);

    //fireEvent dispara el evento los metodos que tiene corresponden a los diferentes tipos de eventos y adentro del parentesis especificamos el lugar en el que ocurre el evento.
    fireEvent.click(screen.getByText("+"));
    //Aqui pruebo lo que espero recibir del evento.
    expect(screen.getByText("19")).toBeTruthy();
  });

  test("debe de decrementar con el boton -", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("-"));
    expect(screen.getByText(17)).toBeTruthy();
  });

  test("debe de hacer reset con el boton Reset", () => {
    //sujeto de prueba
    render(<CounterApp value={value} />);
    //Aqui simulo el evento de click en el que le somo y cambio el estado del counter para luego probar el evento del reset a ver si coincide realmente con el valor inicial que le colocamos.
    //
    //accion de Estimulo
    fireEvent.click(screen.getByText("+"));
    fireEvent.click(screen.getByText("+"));
    fireEvent.click(screen.getByText("+"));
    fireEvent.click(screen.getByText("Reset"));
    //Asercion de comportamiento esperado
    expect(screen.getByText(18)).toBeTruthy();
  });
});
