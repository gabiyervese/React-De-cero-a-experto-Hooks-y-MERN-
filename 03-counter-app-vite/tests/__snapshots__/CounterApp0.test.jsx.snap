// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas basicas del componenete <CounterApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2
    data-testid="test-title"
  />
  <h3 />
  <p>
    6
  </p>
  <button>
    +
  </button>
  <button>
    -
  </button>
  <button>
    Reset
  </button>
</div>
`;
